data-prepper-plugins/kinesis-source/src/main/java/org/opensearch/dataprepper/plugins/kinesis/source/processor/KinesisRecordProcessor.java

```diff
@@ -177,6 +191,7 @@ public void processRecords(ProcessRecordsInput processRecordsInput) {
             // Flush buffer at the end
             bufferAccumulator.flush();
             recordsProcessed.increment(eventCount);
+            bytesProcessedSummary.record(recordBytes.stream().mapToLong(Integer::longValue).sum());
```

Comment:
I think we should record each record independently. This will allow us to know the record size as well.

```
processRecordsInput.records()
  .stream()
  .map(kinesisClientRecord.data().remaining())
  .map(Integer::longValue)
  .forEach(bytesProcessedSummary::record);
```